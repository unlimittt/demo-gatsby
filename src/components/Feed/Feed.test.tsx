import React from "react"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"
import Feed from "./Feed"

Enzyme.configure({ adapter: new Adapter() })

describe("Feed", () => {
  const props = {
    edges: [
      {
        node: {
          fields: {
            slug: "/test_0",
            categorySlug: "/test_0",
            tagSlugs: ["/test-1", "/test-2"],
          },
          frontmatter: {
            date: "2016-09-01",
            description: "test_0",
            category: "test_0",
            tags: ["test-1", "test-2"],
            title: "test_0",
          },
          id: "test-123",
          html: "<p>test</p>",
        },
      },
      {
        node: {
          fields: {
            slug: "/test_1",
            categorySlug: "/test_1",
            tagSlugs: ["/test-1", "/test-2"],
          },
          frontmatter: {
            date: "2016-09-01",
            description: "test_1",
            category: "test_1",
            tags: ["test-1", "test-2"],
            title: "test_1",
          },
          id: "test-321",
          html: "<p>test</p>",
        },
      },
    ],
  }

  it("renders correctly", () => {
    const wrapper = shallow(<Feed {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
