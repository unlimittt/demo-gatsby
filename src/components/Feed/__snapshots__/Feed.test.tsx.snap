// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
<div
  className="feed"
>
  <div
    className="feed__item"
    key="/test_0"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <mockConstructor
          className="feed__item-meta-category-link"
          to="/test_0"
        >
          test_0
        </mockConstructor>
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      <mockConstructor
        className="feed__item-title-link"
        to="/test_0"
      >
        test_0
      </mockConstructor>
    </h2>
    <p
      className="feed__item-description"
    >
      test_0
    </p>
    <mockConstructor
      className="feed__item-readmore"
      to="/test_0"
    >
      Read
    </mockConstructor>
  </div>
  <div
    className="feed__item"
    key="/test_1"
  >
    <div
      className="feed__item-meta"
    >
      <time
        className="feed__item-meta-time"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="feed__item-meta-divider"
      />
      <span
        className="feed__item-meta-category"
      >
        <mockConstructor
          className="feed__item-meta-category-link"
          to="/test_1"
        >
          test_1
        </mockConstructor>
      </span>
    </div>
    <h2
      className="feed__item-title"
    >
      <mockConstructor
        className="feed__item-title-link"
        to="/test_1"
      >
        test_1
      </mockConstructor>
    </h2>
    <p
      className="feed__item-description"
    >
      test_1
    </p>
    <mockConstructor
      className="feed__item-readmore"
      to="/test_1"
    >
      Read
    </mockConstructor>
  </div>
</div>
`;
